### ISSUE TITLE
Get-ModuleVersion: Export this helper function

### ISSUE THREAD

**Issue by johlju** [gaelcolas/Sampler#141]
This function need to be used by for example external tasks to evaluate the module version correctly, and without duplicating the code. We should export this function, either from this module or a another module.

https://github.com/gaelcolas/Sampler/blob/4011fefe903f4515de4d2b8896549a6464a66b0e/.build/tasks/Common.Functions.psm1#L69-L112

**Comment by johlju** [comment]
Comment-based help and unit tests exist here that can be used https://github.com/dsccommunity/DscResource.DocGenerator/blob/master/source/Private/Get-ModuleVersion.ps1

**Comment by johlju** [comment]
We should also export `Get-ProjectName` and `Get-SourcePath`.

**Comment by johlju** [comment]
There is also comment-based help for `Split-ModuleVersion`

**Comment by johlju** [comment]
Saving unit tests her that was removed from DscResource.DocGenerator.
```powershell
Describe 'Get-ModuleVersion' {
        Context 'When ModuleVersion is not $null' {
            It 'Should return the correct module version' {
                Get-ModuleVersion -ModuleVersion '1.0.0-preview1' | Should -Be '1.0.0-preview1'
            }

            It 'Should return the correct module version when it contained metadata' {
                Get-ModuleVersion -ModuleVersion '1.0.0-preview1+metadata.2019-01-01' | Should -Be '1.0.0-preview1'
            }
        }

        Context 'When ModuleVersion is $null' {
            Context 'When module manifest has a prerelease string set' {
                BeforeAll {
                    Mock -CommandName Import-PowerShellDataFile -MockWith {
                        return @{
                            ModuleVersion = '2.0.0'
                            PrivateData = @{
                                PSData = @{
                                    Prerelease = 'preview2'
                                }
                            }
                        }
                    }
                }

                It 'Should return the correct module version' {
                    Get-ModuleVersion -OutputDirectory $TestDrive -ProjectName 'MyModule' | Should -Be '2.0.0-preview2'
                }
            }

            Context 'When module manifest does not have prerelease string set' {
                BeforeAll {
                    Mock -CommandName Import-PowerShellDataFile -MockWith {
                        return @{
                            ModuleVersion = '2.0.0'
                            PrivateData = @{
                                PSData = @{
                                    Prerelease = ''
                                }
                            }
                        }
                    }
                }

                It 'Should return the correct module version' {
                    Get-ModuleVersion -OutputDirectory $TestDrive -ProjectName 'MyModule' | Should -Be '2.0.0'
                }
            }
        }

    }
````

**Comment by gaelcolas** [comment]
This should get fixed by #238

**Comment by gaelcolas** [comment]
The following functions are now public:
```text
Build-Module.ModuleBuilder.build.Sampler.ib.tasks
Invoke-Pester.pester.build.Sampler.ib.tasks
Merge-CodeCoverageFiles.pester.build.Sampler.ib.tasks
Set-BuildEnvironment.BuildHelpers.build.Sampler.ib.tasks
Add-Sample
Get-BuildVersion
Get-BuiltModuleVersion
Get-ClassBasedResourceName
Get-CodeCoverageThreshold
Get-MofSchemaName
Get-OperatingSystemShortName
Get-PesterOutputFileFileName
Get-SamplerCodeCoverageOutputFile
Get-SamplerCodeCoverageOutputFileEncoding
Get-SamplerProjectName
Get-SamplerSourcePath
Merge-JaCoCoReport
New-SampleModule
Update-JaCoCoStatistic
```

Please raise new issues or send PR if you'd like this to change.
